pkgs/dart_tooling_mcp_server/lib/src/server.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:dart_mcp/server.dart';
+import 'package:dtd/dtd.dart';
+import 'package:stream_channel/stream_channel.dart';
+import 'package:vm_service/vm_service_io.dart';
+
+import 'dtd.dart';
+
+/// An MCP server that is connected to the Dart Tooling Daemon (see
+/// https://pub.dev/packages/dtd).
+class DartToolingMCPServer extends MCPServer with ToolsSupport {
+  @override
+  final implementation = ServerImplementation(
+    name: 'dart tooling daemon',
+    version: '0.1.0-wip',
+  );
+
+  @override
+  final instructions =
+      'This server helps to connect Dart and Flutter developers to their '
+      'running apps.';
```

Comment:
not just running apps, but really to dart & flutter developer tools, including access to their running app.